src/libertem/web/connect.py

```diff
@@ -57,23 +61,31 @@ def initialize(self, state: SharedState, event_registry):
                     client_kwargs=client_kwargs
                 )
             )
+            connected = True
         else:
             raise ValueError("unknown connection type")
-        executor = AsyncAdapter(wrapped=sync_executor)
-        await self.state.executor_state.set_executor(executor, request_data)
-        await self.state.dataset_state.verify()
-        datasets = await self.state.dataset_state.serialize_all()
-        msg = Message(self.state).initial_state(
-            jobs=self.state.job_state.serialize_all(),
-            datasets=datasets, analyses=self.state.analysis_state.serialize_all(),
-            compound_analyses=self.state.compound_analysis_state.serialize_all(),
-        )
-        log_message(msg)
-        # FIXME: don't broadcast, only send to the websocket that matches this HTTP connection
-        # (is this even possible?)
-        self.event_registry.broadcast_event(msg)
-        await self.send_existing_job_results()
-        self.write({
-            "status": "ok",
-            "connection": connection,
-        })
+        if connected:
+            executor = AsyncAdapter(wrapped=sync_executor) 
+            await self.state.executor_state.set_executor(executor, request_data)
+            await self.state.dataset_state.verify()
+            datasets = await self.state.dataset_state.serialize_all()
+            msg = Message(self.state).initial_state(
+                jobs=self.state.job_state.serialize_all(),
+                datasets=datasets, analyses=self.state.analysis_state.serialize_all(),
+                compound_analyses=self.state.compound_analysis_state.serialize_all(),
+            )
+            log_message(msg)
+            # FIXME: don't broadcast, only send to the websocket that matches this HTTP connection
+            # (is this even possible?)
+            self.event_registry.broadcast_event(msg)
+            await self.send_existing_job_results()
+            self.write({
+                "status": "ok",
+                "connection": connection,
+            })
+        else:
+            self.write({
+                "status": "disconnected",
+                "connection": {},
+            })
```

Comment:
This could (should?) also return the concrete error message from the exception - possibly also using `"status": "error"` instead? See also `messages.py`, where several `_error` messages are defined. If you like, you can also extract the response into a method of the `Messages` class, so it won't be forgotten once the schema project is implemented.